modules/services/shpool.nix

```diff
@@ -0,0 +1,99 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    getExe
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    ;
+
+  cfg = config.services.shpool;
+  format = pkgs.formats.toml { };
+  configFile = format.generate "config.toml" cfg.settings;
```

Comment:
Would rather inline this, it has 1 usage and involves jumping to the top of the file to see what you're passing into the config source.